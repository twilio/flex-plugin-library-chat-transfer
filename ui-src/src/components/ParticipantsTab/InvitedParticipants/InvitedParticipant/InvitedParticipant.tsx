import { Flex as FlexBox, Box, Button } from '@twilio-paste/core';
import { ChatIcon } from '@twilio-paste/icons/cjs/ChatIcon';
import { AgentIcon } from '@twilio-paste/icons/cjs/AgentIcon';
import { CloseIcon } from '@twilio-paste/icons/cjs/CloseIcon';
import { useState } from 'react';
import React from 'react';

import { ParticipantInviteType } from '../../../../flex-hooks/types/ParticipantInvite';

interface InvitedParticipantProps {
  participantName: string;
  inviteTargetType: ParticipantInviteType;
  handleCancelInvite: () => void;
}

export const InvitedParticipant = ({
  participantName,
  inviteTargetType,
  handleCancelInvite,
}: InvitedParticipantProps) => {
  const [cancelHandled, setCancelHandled] = useState(false);

  const handleCancel = () => {
    setCancelHandled(true);
    handleCancelInvite();
  };

  const icon =
    inviteTargetType === 'Worker' ? (
      <AgentIcon decorative={false} title="ConvTransferAgent" />
    ) : (
      <ChatIcon decorative={false} title="ConvTransferQueue" />
    );

  return (
    <FlexBox>
      <FlexBox>
        <Box padding="space20">{icon}</Box>
      </FlexBox>
      <FlexBox grow>
        <Box padding="space20" width="100%">
          {participantName}
        </Box>
      </FlexBox>
      <FlexBox>
        <Box padding="space20">
          <Button variant="secondary" size="icon" disabled={cancelHandled} onClick={handleCancel}>
            <CloseIcon decorative title="ConvTransferCancelInvite" />
          </Button>
        </Box>
      </FlexBox>
    </FlexBox>
  );
};
