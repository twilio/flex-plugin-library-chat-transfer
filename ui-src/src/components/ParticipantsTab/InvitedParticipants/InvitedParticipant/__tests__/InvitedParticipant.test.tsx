import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { InvitedParticipant } from '../InvitedParticipant';

describe('InvitedParticipant', () => {
  const mockParticipantName = 'John Doe';
  const mockInviteTargetType = 'Worker';
  const mockHandleCancelInvite = jest.fn();

  it('renders correctly with Worker invite type', () => {
    const { getByText, getByTitle } = render(
      <InvitedParticipant
        participantName={mockParticipantName}
        inviteTargetType={mockInviteTargetType}
        handleCancelInvite={mockHandleCancelInvite}
      />,
    );

    const cancelButton = getByTitle('ConvTransferCancelInvite');

    fireEvent.click(cancelButton);

    expect(mockHandleCancelInvite).toHaveBeenCalledTimes(1);
  });

  it('renders screenshot', () => {
    const wrapper = render(
      <InvitedParticipant
        participantName={mockParticipantName}
        inviteTargetType={mockInviteTargetType}
        handleCancelInvite={mockHandleCancelInvite}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  // Add more test cases for other invite target types if needed
});
