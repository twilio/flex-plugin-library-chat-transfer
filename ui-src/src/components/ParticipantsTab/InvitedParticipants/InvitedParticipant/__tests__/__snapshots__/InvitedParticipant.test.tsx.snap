// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitedParticipant renders screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style>
      
    table.disable-selection {
      -webkit-touch-callout: none;
      -webkit-user-select: none; 
      -khtml-user-select: none; 
      -moz-user-select: none; 
      -ms-user-select: none; 
      user-select: none;
    }
  
    .disable-selection span::selection{
      background-color: transparent;
    }
    .disable-selection br::selection{
      background-color: transparent;
    }
  
    </style>
    <div>
      <div
        class="css-1cmiaub"
        data-paste-element="FLEX"
      >
        <div
          class="css-1cmiaub"
          data-paste-element="FLEX"
        >
          <div
            class="css-1vdm3l6"
            data-paste-element="BOX"
          >
            <span
              class="css-144s81u"
              data-paste-element="ICON"
            >
              <svg
                aria-hidden="false"
                aria-labelledby="AgentIcon-5"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="AgentIcon-5"
                >
                  ConvTransferAgent
                </title>
                <path
                  d="M13.484 2.483a6.845 6.845 0 013.336 5.663c.492.106.941.389 1.252.813.333.454.473 1.023.388 1.58l-.414 2.738a2.105 2.105 0 01-2.477 1.768 6.831 6.831 0 01-3.943 2.427 1.842 1.842 0 01-3.493-.835 1.85 1.85 0 011.845-1.856 1.85 1.85 0 011.821 1.555 5.758 5.758 0 002.787-1.766 2.132 2.132 0 01-.709-1.927l.414-2.738a2.104 2.104 0 011.573-1.744l-.104.03a5.785 5.785 0 00-11.564-.013A2.1 2.1 0 015.708 9.9l.415 2.743a2.133 2.133 0 01-.388 1.58 2.105 2.105 0 01-3.78-.942l-.415-2.742a2.129 2.129 0 01.387-1.58c.302-.412.734-.69 1.21-.804a6.845 6.845 0 0110.347-5.673zM9.978 15.84a.79.79 0 00-.785.796c0 .44.353.795.785.795a.79.79 0 00.785-.795.79.79 0 00-.785-.796zM3.754 9.166a1.044 1.044 0 00-.971.42 1.069 1.069 0 00-.195.794l.416 2.746a1.044 1.044 0 001.875.47c.168-.229.238-.514.195-.794l-.416-2.748a1.044 1.044 0 00-.78-.865zm13.462.42a1.044 1.044 0 00-1.876.473l-.414 2.743c-.043.28.027.565.194.793a1.044 1.044 0 001.877-.473l.414-2.743a1.069 1.069 0 00-.195-.793z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="css-j3kwsk"
          data-paste-element="FLEX"
        >
          <div
            class="css-1ovf07"
            data-paste-element="BOX"
          >
            John Doe
          </div>
        </div>
        <div
          class="css-1cmiaub"
          data-paste-element="FLEX"
        >
          <div
            class="css-1vdm3l6"
            data-paste-element="BOX"
          >
            <button
              aria-busy="false"
              class="css-xdezt4"
              data-paste-element="BUTTON"
              type="button"
            >
              <span
                class="css-z6jp9v"
                data-paste-element="BOX"
              >
                <span
                  class="css-144s81u"
                  data-paste-element="ICON"
                >
                  <svg
                    aria-hidden="true"
                    aria-labelledby="CloseIcon-6"
                    height="100%"
                    role="img"
                    viewBox="0 0 20 20"
                    width="100%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="CloseIcon-6"
                    >
                      ConvTransferCancelInvite
                    </title>
                    <path
                      d="M15.16 13.514L11.645 10l3.515-3.514a1.165 1.165 0 00-1.646-1.646L10 8.355 6.486 4.84A1.165 1.165 0 004.84 6.486L8.355 10 4.84 13.514a1.165 1.165 0 001.646 1.646L10 11.645l3.514 3.515a1.165 1.165 0 001.646-1.646z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1cmiaub"
      data-paste-element="FLEX"
    >
      <div
        class="css-1cmiaub"
        data-paste-element="FLEX"
      >
        <div
          class="css-1vdm3l6"
          data-paste-element="BOX"
        >
          <span
            class="css-144s81u"
            data-paste-element="ICON"
          >
            <svg
              aria-hidden="false"
              aria-labelledby="AgentIcon-5"
              height="100%"
              role="img"
              viewBox="0 0 20 20"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="AgentIcon-5"
              >
                ConvTransferAgent
              </title>
              <path
                d="M13.484 2.483a6.845 6.845 0 013.336 5.663c.492.106.941.389 1.252.813.333.454.473 1.023.388 1.58l-.414 2.738a2.105 2.105 0 01-2.477 1.768 6.831 6.831 0 01-3.943 2.427 1.842 1.842 0 01-3.493-.835 1.85 1.85 0 011.845-1.856 1.85 1.85 0 011.821 1.555 5.758 5.758 0 002.787-1.766 2.132 2.132 0 01-.709-1.927l.414-2.738a2.104 2.104 0 011.573-1.744l-.104.03a5.785 5.785 0 00-11.564-.013A2.1 2.1 0 015.708 9.9l.415 2.743a2.133 2.133 0 01-.388 1.58 2.105 2.105 0 01-3.78-.942l-.415-2.742a2.129 2.129 0 01.387-1.58c.302-.412.734-.69 1.21-.804a6.845 6.845 0 0110.347-5.673zM9.978 15.84a.79.79 0 00-.785.796c0 .44.353.795.785.795a.79.79 0 00.785-.795.79.79 0 00-.785-.796zM3.754 9.166a1.044 1.044 0 00-.971.42 1.069 1.069 0 00-.195.794l.416 2.746a1.044 1.044 0 001.875.47c.168-.229.238-.514.195-.794l-.416-2.748a1.044 1.044 0 00-.78-.865zm13.462.42a1.044 1.044 0 00-1.876.473l-.414 2.743c-.043.28.027.565.194.793a1.044 1.044 0 001.877-.473l.414-2.743a1.069 1.069 0 00-.195-.793z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="css-j3kwsk"
        data-paste-element="FLEX"
      >
        <div
          class="css-1ovf07"
          data-paste-element="BOX"
        >
          John Doe
        </div>
      </div>
      <div
        class="css-1cmiaub"
        data-paste-element="FLEX"
      >
        <div
          class="css-1vdm3l6"
          data-paste-element="BOX"
        >
          <button
            aria-busy="false"
            class="css-xdezt4"
            data-paste-element="BUTTON"
            type="button"
          >
            <span
              class="css-z6jp9v"
              data-paste-element="BOX"
            >
              <span
                class="css-144s81u"
                data-paste-element="ICON"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="CloseIcon-6"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="CloseIcon-6"
                  >
                    ConvTransferCancelInvite
                  </title>
                  <path
                    d="M15.16 13.514L11.645 10l3.515-3.514a1.165 1.165 0 00-1.646-1.646L10 8.355 6.486 4.84A1.165 1.165 0 004.84 6.486L8.355 10 4.84 13.514a1.165 1.165 0 001.646 1.646L10 11.645l3.514 3.515a1.165 1.165 0 001.646-1.646z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
